storage/testdb/testdb.go

```diff
@@ -35,25 +35,26 @@ import (
 )
 
 const (
-	// MySQLURIEnv is the name of the ENV var checked for the test MySQL
-	// instance URI to use.
+	// MySQLURIEnv is the name of the ENV variable checked for the test MySQL
+	// instance URI to use. The value must have a trailing slash.
 	MySQLURIEnv = "TEST_MYSQL_URI"
 
-	defaultTestMySQLURI = "root@tcp(127.0.0.1)"
+	// Note: sql.Open requires the URI to end with a slash.
+	defaultTestMySQLURI = "root@tcp(127.0.0.1)/"
 )
 
 var (
 	trillianSQL = testonly.RelativeToPackage("../mysql/schema/storage.sql")
 )
 
-// mysqlURI returns the connection URI to use for tests.
-// It will prefer the value in the MYSQL_URI env var, and if empty/unset, it'll
-// return the defaultTestMySQLURI value.
+// mysqlURI returns the MySQL connection URI to use for tests. It returns the
+// value in the ENV variable defined by MySQLURIEnv. If the value is empty,
+// returns defaultTestMySQLURI.
 //
-// Only a subset of the suite of tests in this repo require a database it's
-// not possible to blanket apply the test_mysql_uri flag, and maintaining a separate
-// list of tests which needs DB connectivity is brittle, so this fallback approach
-// means we can provide the connection info in a more ambient fashion.
+// We use an ENV variable, rather than a flag, for flexibility. Only a subset
+// of the tests in this repo require a database and import this package. With a
+// flag, it would be necessary to distinguish "go test" invocations that need a
+// database, and those that don't. ENV allows to "blanket apply" this setting.
 func mysqlURI() string {
```

Comment:
as per the other comments in the file - perhaps mysqlURI() should check and guarantee that what it returns ends with a `/` so that all sql.Open(mysqlURI()) calls are covered?

Reply:
If it doesn't end with a slash, `sql.Open` will return an error, the tests will fail, and the caller will add a slash. IMO, there is no much need in trying to prevent this. It would be a problem if `sql.Open` didn't fail in this case, but silently do some evil.